src/ray/common/runtime_env_manager.cc

```diff
@@ -51,21 +58,23 @@ const std::vector<std::string> &RuntimeEnvManager::GetReferences(
 
 void RuntimeEnvManager::RemoveURIReference(const std::string &hex_id) {
   RAY_LOG(DEBUG) << "Subtracting 1 from URI Reference for id " << hex_id;
-  if (!id_to_uris_.count(hex_id)) {
+  auto iter = id_to_uris_.find(hex_id);
+  if (iter == id_to_uris_.end()) {
     return;
   }
 
-  for (const auto &uri : id_to_uris_[hex_id]) {
-    --uri_reference_[uri];
-    auto ref_count = uri_reference_[uri];
-    RAY_CHECK(ref_count >= 0);
-    if (ref_count == 0) {
-      uri_reference_.erase(uri);
+  for (const auto &uri : iter->second) {
+    auto uri_ref_iter = uri_reference_.find(uri);
+    RAY_CHECK(uri_ref_iter != uri_reference_.end());
+    const auto new_ref_count = --uri_ref_iter->second;
```

Comment:
I don't like -- and ++ with assignments. Let's put it 2 lines

Reply:
> I don't like -- and ++ with assignments

Any reasoning? Coding practice / suggestion?

Reply:
code readability. when I read 
```
auto y = ++x;
```

I need to reason: okay so y is assigned to the value before increment, not after. 

but when I read

```
auto y = x;
++x;
```

there's no such cognitive load.

Reply:
wait I even made a mistake, it's after the increment not before lol. this really makes the point.

Reply:
I think this is fundamental C/C++ grammar, not something fancy and hard-to-understand.
But updated to make you happy.